bug: (go1.27) unusedmethod analyzer false positive with generic methods

go1.27 will introduce generic methods.

`unusedmethod` analyzer is not working as expected with those new elements.

<details>
<summary>main.go</summary>

```go
package main

func main() {
	var g G[string]
	var _ I = g

	g.m("foo")

	var h H
	h.m[string]("foo")
}

type G[P any] struct{}

func (G[P]) m(P) {}

type I interface {
	m(string)
}

type H struct{}

func (H) m[P any](P) {}

```

</details>

<details>

```yml
version: 2

linters:
  default: none
  enable:
    - iface
  settings:
    iface:
      enable:
        - unusedmethod
```

</details>

```console
$ ./golangci-lint run
main.go:18:2: unusedmethod: method 'm()' is declared on interface 'I' but not used within the package (iface)
        m(string)
        ^
1 issues:
* iface: 1
```

--- a/unusedmethod/doc.go
+++ b/unusedmethod/doc.go
@@ -1,3 +1,8 @@
 // Package unusedmethod defines an Analyzer that detects interface methods which
 // are never used anywhere in the same package where they are defined.
+//
+// A method is considered used only when it is invoked or referenced through a
+// value of the interface type. Merely implementing the interface — for example,
+// assigning a concrete type to the interface, or calling the method directly on
+// the concrete type — does not count as a use.
 package unusedmethod

--- a/unusedmethod/testdata/src/concrete/main.go
+++ b/unusedmethod/testdata/src/concrete/main.go
@@ -1,0 +1,23 @@
+package main
+
+func main() {
+	var g G
+	var _ I = g
+
+	g.m("foo")
+
+	var h H
+	h.m("foo")
+}
+
+type G struct{}
+
+func (G) m(string) {}
+
+type I interface {
+	m(string) // want "^method 'm\\(\\)' is declared on interface 'I' but not used within the package$"
+}
+
+type H struct{}
+
+func (H) m(string) {}

--- a/unusedmethod/unusedmethod.go
+++ b/unusedmethod/unusedmethod.go
@@ -22,7 +22,7 @@ func newAnalyzer() *analysis.Analyzer {
 
 	analyzer := &analysis.Analyzer{
 		Name:     "unusedmethod",
-		Doc:      "Detects interface methods that are never used anywhere in the same package where they are defined.",
+		Doc:      "Detects interface methods that are never used anywhere in the same package where they are defined. A method is considered used only when invoked or referenced through a value of the interface type; merely implementing the interface does not count as a use.",
 		URL:      "https://pkg.go.dev/github.com/uudashr/iface/unusedmethod",
 		Requires: []*analysis.Analyzer{inspect.Analyzer},
 		Run:      r.run,

--- a/unusedmethod/unusedmethod_test.go
+++ b/unusedmethod/unusedmethod_test.go
@@ -20,5 +20,6 @@ func Test(t *testing.T) {
 	analysistest.Run(t, testdata, unusedmethod.Analyzer,
 		"excludepkg",
 		"methodref",
-		"methodexpr")
+		"methodexpr",
+		"concrete")
 }

